### ISSUE TITLE
amplify-authenticator component shouldn't render default slot when not authenticated

### ISSUE THREAD

**Issue by srchulo** [aws-amplify/amplify-js#6087]
**Is your feature request related to a problem? Please describe.**
Currently when using the `amplify-authenticator` Vue component, the inner content will be hidden, but still be evaluated. This can lead to API requests failing, but also weird behaviors with lifecycle methods like created because the component has been created, even if it hasn't been shown yet.

**Describe the solution you'd like**
It would be nice if the default slot actually wasn't rendered until the user was authenticated.

**Describe alternatives you've considered**

I've written this in my default Nuxt layout to accomplish what I want:

```
<template>
    <amplify-authenticator>
        <amplify-sign-in slot="sign-in" username-alias="email"></amplify-sign-in>

        <nuxt v-if="signedIn" />
        <amplify-sign-out></amplify-sign-out>
    </amplify-authenticator>
</template>

<script>
import { Auth, Hub } from 'aws-amplify'

export default {
    data() {
        return {
            signedIn: null,
        }
    },
    watch: {
        signedIn() {
            if (!this.signedIn) {
                this.$router.push('/')
            }
        },
    },
    created() {
        const component = this
        Hub.listen('auth', async ({ payload }) => {
            switch (payload.event) {
                case 'signIn':
                    component.signedIn = true
                    break
                case 'signOut':
                case 'signIn_failure':
                    component.signedIn = false
                    break
                case 'configured':
                    try {
                        await Auth.currentAuthenticatedUser()
                        this.signedIn = true
                    } catch (e) {
                        this.signedIn = false
                    }
            }
        })
    },
}
</script>
```

**Comment by jordanranz** [comment]
Yes, this is a limitation with how the default slot work for web components in Stencil. I am unsure if we will be able to get around this limitation so I have documented the recommended way to work around this: https://docs.amplify.aws/ui/auth/authenticator/q/framework/vue#manage-auth-state-and-conditional-app-rendering. 

This documentation will highlight how to manage auth state and app rendering within your app similar to what you have already done but will use the Authenticator's Auth state and a simplified `onAuthUIStateChange` function that wraps the Hub listener auth UI event management.

Let us know if you have any feedback for this.

**Comment by srchulo** [comment]
I think that documentation helps-- example is simpler than mine!

One question, now `amplify-authenticator` doesn't wrap the rest of the content. I still left that around the content to be safe, but I guess in your example you're saying that it's not needed because the `onAuthUIStateChange` listener is effectively doing the same thing?

Also, I think the example is nice with a computed property:

```
    computed: {
        signedIn() {
            return this.authState === 'signedin' && this.user
        },
    }
```

Then these can be simplified to:

```
<amplify-authenticator v-if="!signedIn'"></amplify-authenticator>
<div v-if="signedIn">
    ...
</div>
```

And could you post a link to the documentation for `onAuthUIStateChange`? Sorry-- I tried to find it, but couldn't!

**Comment by srchulo** [comment]
Also, it might be nice if the documentation explained why `onAuthUIStateChange` is returned in beforeDestroy, unless that documentation is already covered wherever `onAuthUIStateChange` is documented?
